exporter/exporter.go

```diff
@@ -559,7 +560,16 @@ func (e *Exporter) scrapeRedisHost(ch chan<- prometheus.Metric) error {
 
 	e.extractLatencyMetrics(ch, c)
 
-	e.extractCheckKeyMetrics(ch, c)
+	if e.options.IsCluster {
+		clusterClient, err := e.connectToRedisCluster()
```

Comment:
1.  Don't forget to call `defer()` when creating any connection:
```go
defer clusterClient.Close()
```
2. Is it ok to use 2 separate clients/connections in the single `scareHost()`?

Reply:
1. that is very good catch, adding the `defer close()` after line 568 should do the trick.

2. I think that's fine.

Reply:
I have added the defer in the function. Good catch @houstonheat . Thank you.